utils/calendrical_calculations/Cargo.toml

```diff
@@ -23,7 +23,7 @@ rust-version.workspace = true
 
 # This is a special exception: The algorithms in this crate are based on "Calendrical Calculations" by Reingold and Dershowitz
 # which has its lisp code published at https://github.com/EdReingold/calendar-code2/
-license = "Apache-2.0"
+license = "Apache-2.0" # TODO: Need to add `MIT`/`GNU`?
```

Comment:
No, the license was picked by a lawyer. Why would GNU be needed?

Reply:
[There is](https://github.com/cassioneri/eaf/blob/main/algorithms/neri_schneider.hpp) a realization of the algorithm(from the author of the article) in C/C++ and [in the repo no](https://github.com/cassioneri/eaf/tree/main/LICENSES) apache 2.0 license

[Here](https://github.com/unicode-org/icu4x/blob/0d46522d0a78e575abbacf1fce2ad692dbf0f113/utils/calendrical_calculations/src/iso.rs#L3-L8) is a comment with mentioning it in the PR

So I don't sure is it need to be added any of them or not

Reply:
Ah. We'd have to talk to the lawyer again for this.

Typically algorithms themselves aren't copyrightable, however we would indeed need to check with our lawyer to pull in this code.

Reply:
I believe the policy is, we can redistribute MIT (but not GPL) code under the Apache-2.0 license, and all third-party code should retain its copyright comments inline in the code, similar to the Reingold code.

Reply:
The first link goes to a file with the following comment at the top

```
// SPDX-License-Identifier: GPL-3.0-or-later
```

That might mean that we would not be able with our current licenses to consume this code in the ICU4X project, because the ICU4X project does not distribute itself under the GPL license, and furthermore, some of our customers do not accept GPL licensed code. However, if there are portions of the code that are MIT licensed instead of GPL licensed, we could accept those.

Reply:
A workaround to this type of issue would be for any GPL-licensed code to live in its own crate, and then either `icu_calendar` or `calendrical_calculations` has an optional Cargo feature to consume it. Clients who would like the speedup and are okay consuming GPL-licensed code would need to manually enable the Cargo feature.

I do not know whether such a GPL crate could live in this repository or whether it would need its own repository.

Reply:
I'd rather avoid introducing GPL licensed code in our dep tree at all.

Reply:
@sffc @Manishearth 
As I can understand you can implement algos from an article without such license restriction.
And if you will check the code by link and the code from PR it's will be pretty clear* that code was inspired by the article and only then it was matched with author's code for reference to authority. So maybe I can just remove link to the author implementation and leave only links to the article?

[*]: because of naming(unnamed const and very short names that say almost nothing (except for y/m/d ones, yeah they can say something, but nothing about how and why)) in the repo of the article's author. And in the PR even some consts was changed because of we have larger valid dates' interval -- in the author's code they are just magic numbers again; and how will you change such consts without understanding for what and why? And of course in the PR code there is plenty comments about why and how.

Reply:
Yes, I understand that generally algorithms are not copyrightable, but either way, we will have to get approval from our lawyer for doing this, and they may choose to be more cautious about this. We were already quite cautious about the Reingold&Dershowitz algoritms.